sdk/core/System.ClientModel/src/Convenience/AsyncEnumerableResultOfT.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace System.ClientModel;
+
+#pragma warning disable CS1591 // public XML comments
+public abstract class AsyncEnumerableResult<T> : ClientResult, IAsyncEnumerable<T>, IDisposable
```

Comment:
Inheriting from ClientResult adds the `GetRawResponse` method to this type.  For enumerable client result implementations that use a polling mechanism and therefore have multiple responses associated with them, we might need to either say "GetRawResponse returns the first response and subtypes must expose other public APIs to access subsequent responses," or possibly expose a `GetRawResponses` method that returns a collection of PipelineResponse values.  The latter is not a great Dev UX in the streaming response case where there is only a single response returned, but the former could require us to add additional types to support polling-enumerable scenarios such as paging.